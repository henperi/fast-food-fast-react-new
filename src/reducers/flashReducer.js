import { SHOW_FLASH, REMOVE_FLASH } from '../actions/actionTypes';

const initialState = {
  showFlash: false,
  title: '',
  message: '',
  flashType: 'success',
};

const flashReducer = (state = initialState, action) => {
  switch (action.type) {
    case SHOW_FLASH:
      return {
        showFlash: true,
        ...action.payload,
      };
    case REMOVE_FLASH:
      return {
        showFlash: false,
      };

    default:
      return state;
  }
};

export default flashReducer;
