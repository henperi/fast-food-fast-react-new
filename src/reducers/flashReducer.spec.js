/* eslint-disable no-undef */
import flashReducer, { initialState } from './flashReducer';

describe('flashReducer', () => {
  // const initialState = [];
  it('should return the default state if no action is specified', () => {
    const action = { type: '' };
    expect(flashReducer(initialState, action)).toMatchSnapshot({});
    expect(flashReducer(initialState, action)).toEqual({
      flashType: 'success',
      message: '',
      showFlash: false,
      title: '',
    });
  });

  it('should update the state and SHOW_FLASH when type is SET_MENU and payload is specified', () => {
    const action = {
      type: 'SHOW_FLASH',
      payload: {
        flashType: 'success',
        message: 'new message',
        showFlash: true,
        title: 'new title',
      },
    };
    expect(flashReducer(initialState, action)).toMatchSnapshot({});
    expect(flashReducer(initialState, action)).toEqual({
      flashType: 'success',
      message: 'new message',
      showFlash: true,
      title: 'new title',
    });
  });

  it('should update the state and REMOVE_FLASH when type is SET_MENU and payload is specified', () => {
    const action = {
      type: 'REMOVE_FLASH',
    };
    expect(flashReducer(initialState, action)).toMatchSnapshot({});
    expect(flashReducer(initialState, action)).toEqual({ showFlash: false });
  });
});
