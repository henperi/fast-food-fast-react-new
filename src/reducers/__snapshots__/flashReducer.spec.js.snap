// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flashReducer should return the default state if no action is specified 1`] = `
Object {
  "flashType": "success",
  "message": "",
  "showFlash": false,
  "title": "",
}
`;

exports[`flashReducer should update the state and REMOVE_FLASH when type is SET_MENU and payload is specified 1`] = `
Object {
  "showFlash": false,
}
`;

exports[`flashReducer should update the state and SHOW_FLASH when type is SET_MENU and payload is specified 1`] = `
Object {
  "flashType": "success",
  "message": "new message",
  "showFlash": true,
  "title": "new title",
}
`;
